src/sagemaker/utils.py

```diff
@@ -99,9 +100,17 @@ def base_name_from_image(image):
     Returns:
         str: Algorithm name, as extracted from the image name.
     """
-    m = re.match("^(.+/)?([^:/]+)(:[^:]+)?$", image)
-    algo_name = m.group(2) if m else image
-    return algo_name
+    if is_pipeline_variable(image):
+        if is_pipeline_parameter_string(image) and image.default_value:
+            image_str = image.default_value
```

Comment:
The base_name_from_image will appear in s3 uri eventually.
- How about we simply return "base_name" directly in terms of pipeline variable? As the image_uri can be changed in execution time, then s3 uri generated as per image is not making sense and may cause confusions.
- Plus as the base_name_from_image would show up in s3 uri, can we make "base_name" more meaningful? like "processing-job" etc.

Reply:
> How about we simply return "base_name" directly in terms of pipeline variable?

Make sense. However, I wonder the frequency of the change of the `image_uri`, if the underlying algo. does not change, the necessity of change the `image_uri` is not obvious.

> Plus as the base_name_from_image would show up in s3 uri, can we make "base_name" more meaningful? like "processing-job" etc.

Yea we probably should do this. Thought about this before, this would make the code looks ugly, basically, where we call this function (from estimator, processor, model etc), we will need to pass an additional class name, and it is only used if the image name value given is a pipeline variable.